### ISSUE TITLE
Issues with did_action within templates

### ISSUE THREAD

**Issue by SudoCat** [themosis/framework#241]
I'm using the plugin Easy Custom Sidebars (https://en-gb.wordpress.org/plugins/easy-custom-sidebars/), however it's having issues detecting when the wp_head action has been fired.

To replace the sidebars, the plugin first checks `did_action( 'wp_head' )` when the filter `sidebars_widgets` is run. This filter gets called at some point during the `is_active_sidebar` function. The `is_active_sidebar` function exists within my sidebar template, which should definitely get called after `wp_head`. 

After some testing, I've found that `did_action( 'wp_head' )` is returning correctly in my base template and anything included, but is wrong within any of the yield sections. The global `$wp_actions` variable is almost entirely different.

I have absolutely no idea what exactly is happening, any help would be greatly appreciated!

Cheers

**Comment by jlambe** [comment]
If you define scout view files, all sub-views and yield sections are compiled into a "classic" php file which is then use to output everything. The `wp_head` action hooks occurs only if you call the `wp_head()` function inside your theme and must be inside your `<head>` tag.

In theory I can't see why the `$wp_actions` variable is different within sections as it is a global variable totally separated from the views or core WordPress templates. Perhaps check with the plugin author for your sidebars to also test this.

Why not defined your sidebars within the theme config file? What does the plugin do in fact?

**Comment by jlambe** [comment]
You also make sure to use the `@overwrite` statement to close your yield section overwrite in place of `@stop` and check if it works.

**Comment by SudoCat** [comment]
I have checked and the `wp_head` action is definitely being called - the `did_action('wp_head')` returns correctly in everything except `@yield` sections. I just tested using the `@overwrite` statement instead, but it didn't seem to change anything.

The plugin allows the client to easily create new sidebars, and define rules for where they should appear (specific post types, single, archives, taxonomies etc.). This just provides much more dynamic and contextually accurate sidebar content.

I considered contacting the plugin author, but this seems to be more of a bug being caused by Themosis; I've just run through a bunch of my templates, and it seems that all of the yield sections return a different `$wp_actions` array to the base template.

Just in case it helps, here is an example of a `$wp_actions` from within the base template, or any top-level includes. Note that the second is identical, then stops suddenly after the `themosis_run` action.

```
array(51) {
  ["muplugins_loaded"]=>
  int(1)
  ["registered_taxonomy"]=>
  int(10)
  ["registered_post_type"]=>
  int(17)
  ["load_textdomain"]=>
  int(15)
  ["plugins_loaded"]=>
  int(1)
  ["sanitize_comment_cookies"]=>
  int(1)
  ["setup_theme"]=>
  int(1)
  ["unload_textdomain"]=>
  int(1)
  ["after_setup_theme"]=>
  int(1)
  ["themosis_configurations"]=>
  int(1)
  ["themosis_bootstrap"]=>
  int(1)
  ["auth_cookie_malformed"]=>
  int(1)
  ["auth_cookie_valid"]=>
  int(1)
  ["set_current_user"]=>
  int(1)
  ["init"]=>
  int(1)
  ["widgets_init"]=>
  int(1)
  ["wp_register_sidebar_widget"]=>
  int(10)
  ["wp_default_scripts"]=>
  int(1)
  ["wp_default_styles"]=>
  int(1)
  ["acf/include_field_types"]=>
  int(1)
  ["acf/include_fields"]=>
  int(1)
  ["acf/init"]=>
  int(1)
  ["register_sidebar"]=>
  int(2)
  ["parse_tax_query"]=>
  int(17)
  ["parse_query"]=>
  int(9)
  ["pre_get_posts"]=>
  int(9)
  ["posts_selection"]=>
  int(9)
  ["loop_start"]=>
  int(2)
  ["the_post"]=>
  int(4)
  ["loop_end"]=>
  int(2)
  ["wpseo_double_clean_titles"]=>
  int(1)
  ["wp_loaded"]=>
  int(1)
  ["parse_request"]=>
  int(1)
  ["send_headers"]=>
  int(1)
  ["wp"]=>
  int(1)
  ["template_redirect"]=>
  int(1)
  ["admin_bar_init"]=>
  int(1)
  ["add_admin_bar_menus"]=>
  int(1)
  ["themosis_parse_query"]=>
  int(1)
  ["themosis_run"]=>
  int(1)
  ["composing: home.services"]=>
  int(1)
  ["composing: home.carousel"]=>
  int(1)
  ["wp_head"]=>
  int(1)
  ["wpseo_register_extra_replacements"]=>
  int(1)
  ["wp_enqueue_scripts"]=>
  int(1)
  ["wpseo_head"]=>
  int(1)
  ["wpseo_opengraph"]=>
  int(1)
  ["wpseo_twitter"]=>
  int(1)
  ["wpseo_json_ld"]=>
  int(1)
  ["wp_print_styles"]=>
  int(1)
  ["wp_print_scripts"]=>
  int(1)
}
```

And here is the one for any yield sections.
```
array(40) {
  ["muplugins_loaded"]=>
  int(1)
  ["registered_taxonomy"]=>
  int(10)
  ["registered_post_type"]=>
  int(17)
  ["load_textdomain"]=>
  int(15)
  ["plugins_loaded"]=>
  int(1)
  ["sanitize_comment_cookies"]=>
  int(1)
  ["setup_theme"]=>
  int(1)
  ["unload_textdomain"]=>
  int(1)
  ["after_setup_theme"]=>
  int(1)
  ["themosis_configurations"]=>
  int(1)
  ["themosis_bootstrap"]=>
  int(1)
  ["auth_cookie_malformed"]=>
  int(1)
  ["auth_cookie_valid"]=>
  int(1)
  ["set_current_user"]=>
  int(1)
  ["init"]=>
  int(1)
  ["widgets_init"]=>
  int(1)
  ["wp_register_sidebar_widget"]=>
  int(10)
  ["wp_default_scripts"]=>
  int(1)
  ["wp_default_styles"]=>
  int(1)
  ["acf/include_field_types"]=>
  int(1)
  ["acf/include_fields"]=>
  int(1)
  ["acf/init"]=>
  int(1)
  ["register_sidebar"]=>
  int(2)
  ["parse_tax_query"]=>
  int(2)
  ["parse_query"]=>
  int(2)
  ["pre_get_posts"]=>
  int(2)
  ["posts_selection"]=>
  int(2)
  ["loop_start"]=>
  int(1)
  ["the_post"]=>
  int(1)
  ["loop_end"]=>
  int(1)
  ["wpseo_double_clean_titles"]=>
  int(1)
  ["wp_loaded"]=>
  int(1)
  ["parse_request"]=>
  int(1)
  ["send_headers"]=>
  int(1)
  ["wp"]=>
  int(1)
  ["template_redirect"]=>
  int(1)
  ["admin_bar_init"]=>
  int(1)
  ["add_admin_bar_menus"]=>
  int(1)
  ["themosis_parse_query"]=>
  int(1)
  ["themosis_run"]=>
  int(1)
}
```

**Comment by jlambe** [comment]
You don't have any errors thrown from one of your sub-views? it seems odd the actions array stop at routing one. Plus it should be run once from the main view. I'll need to test the plugin with a clean installation and check what's happening.

**Comment by SudoCat** [comment]
I've had a check and can't see any errors, enabled logging and none have appeared in there. Not really sure what's going on to be honest.

**Comment by jlambe** [comment]
Are you running on latest version of the framework? I'll test with the plugin and let you know as I can't figure it out now.

**Comment by SudoCat** [comment]
Hmmm should be, only set up this site at max two weeks ago. Thanks for your help!

**Comment by SudoCat** [comment]
I don't suppose you've had any luck figuring this out? I'm still completely stumped, I'll find another solution if needed.

**Comment by jlambe** [comment]
Hi, I haven't had the time yet. I keep your issue open because it might reveal a bug as well. But yes, I currently suggest you to look for another solution in the meantime. Sorry.

**Comment by SudoCat** [comment]
That's absolutely fine, I just wanted to double check! I'll figure something out.

**Comment by SudoCat** [comment]
I found some time to look further into this problem, it seems like there is something seriously wrong with the order actions are running in. I compared the order of the $wp_actions arrays I'm receiving to the order described on the codex page (https://codex.wordpress.org/Plugin_API/Action_Reference) and it is coming out in a completely different order. I can only presume that something the tempting system is doing is messing with the code execution order.

I'm implementing a more manual solution for the time being.
